src/components/PostDetails.tsx

```diff
@@ -1,106 +1,121 @@
-import React from 'react';
-import { Loader } from './Loader';
+import React, { useEffect } from 'react';
 import { NewCommentForm } from './NewCommentForm';
+import { Loader } from './Loader';
+import { Comment, CommentData } from '../types/Comment';
+import { User } from '../types/User';
+import { Post } from '../types/Post';
+
+type Props = {
+  isCommentLoading: boolean;
+  postComments: Comment[];
+  selectedPost: Post | null;
+  errorsComment: boolean;
+  onDeleteComment: (commentId: number) => void;
+  onAddComment: (newComment: CommentData) => void;
+  isButtonAddLoading: boolean;
+  onShowCommentFormChange: (item: boolean) => void;
```

Comment:
The method `onShowCommentFormChange` doesn't follow the recommended naming conventions for event handlers in React. Consider renaming it to `handleShowCommentFormChange` or `handleCommentFormVisibilityChange`. Check this article for more details: https://medium.com/javascript-in-plain-english/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364